intg/src/main/java/org/apache/atlas/type/AtlasTypeRegistry.java

```diff
@@ -756,13 +757,21 @@ public void removeTypeByName(String name) throws AtlasBaseException {
 
         public List<AtlasBaseTypeDef> getDeleteedTypes() { return deletedTypes; }
 
+        void validateTypeCreation(AtlasBaseTypeDef typeDef) throws AtlasBaseException{
+            if(this.isRegisteredType(typeDef.getName()) && (this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.PRIMITIVE)||(this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.OBJECT_ID_TYPE)))){
```

Comment:
What does "isRegisteredType" check for? Please explain with an example

Reply:
registryData.allTypes contains all types mapped with their typeName. isRegisteredType checks whether a typeName is already present or not in registryData.allTypes.

- if typeName is test1 then isRegisteredType is false as it do not contain it. There will be no error thrown. But if we do not check `this.isRegisteredType(typeDef.getName())` then we’ll get an exception when checking typeCategory as PRIMITIVE/OBJECT_ID_TYPE
<img width="935" alt="image" src="https://user-images.githubusercontent.com/121708100/221908426-bc701d5e-cce3-4ad6-8d63-b840a048271b.png">

- If typeName is date (any name which matches with builtin Type) then isRegisteredType is true because "date" is mapped with its builtin Type.
![image](https://user-images.githubusercontent.com/121708100/221907502-c1455ec4-6c28-4baf-bd77-e5ec6fc77c80.png)

- if typeName is Table (already existing), isRegisteredType is true but typeCategory is not PRIMITIVE/OBJECT_ID_TYPE So it won’t throw forbidden type error. We’ll get “given type already exists”.
![image](https://user-images.githubusercontent.com/121708100/221907653-ca39ab8d-7113-4a6f-8898-d17b475bc687.png)

Reply:
Understood